pkg/backend/filestate/backend.go

```diff
@@ -106,7 +107,18 @@ func New(d diag.Sink, originalURL string) (Backend, error) {
 		return nil, err
 	}
 
-	bucket, err := blob.OpenBucket(context.TODO(), u)
+	blobmux := blob.DefaultURLMux()
+
+	// for gcp we want to support additional credentials
+	// schemes on top of go-cloud's default credentials mux.
+	if strings.HasPrefix(u, gcsblob.Scheme) {
```

Comment:
Nit: I'd prefer that we parse the URL and check the scheme explicitly.

Reply:
@Place1 Do you have time to update this or need a hand?